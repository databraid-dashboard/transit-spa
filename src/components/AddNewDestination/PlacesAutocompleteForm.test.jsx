import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
  PlacesAutocompleteForm,
  mapStateToProps,
  mapDispatchToProps,
} from './PlacesAutocompleteForm';

describe('PlacesAutocompleteForm', () => {
  const state = {
    widgets: {
      byId: {
        transit: {
          configuration: {
            geolocating: true,
            currentLocation: {
              address: '44 Tehama St, San Francisco, CA 94105',
              lat: 37.7873889,
              lng: -122.3964106,
            },
          },
        },
      },
    },
  };

  it('should render an empty form', () => {
    const component = shallow(<PlacesAutocompleteForm />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('mapStateToProps', () => {
    const expected = {
      origin: '44 Tehama St, San Francisco, CA 94105',
    };
    expect(mapStateToProps(state, { widgetId: 'transit' })).toEqual(expected);
  });

  it('mapDispatchToProps', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('addDestination');
  });
});
