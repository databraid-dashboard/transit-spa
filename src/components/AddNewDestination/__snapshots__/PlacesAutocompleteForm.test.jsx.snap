// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesAutocompleteForm should render an empty form 1`] = `
<form
  className="ui semantic"
  onSubmit={[Function]}
>
  <div
    style={
      Object {
        "paddingBottom": "10px",
      }
    }
  >
    <PlacesAutocomplete
      autocompleteItem={[Function]}
      classNames={Object {}}
      clearItemsOnError={false}
      debounce={200}
      googleLogo={true}
      googleLogoType="default"
      highlightFirstSuggestion={false}
      inputProps={
        Object {
          "onChange": [Function],
          "placeholder": "Choose a new destination",
          "value": "600 Guerrero St, San Francisco, CA 94110",
        }
      }
      onError={[Function]}
      options={Object {}}
      styles={Object {}}
    />
  </div>
  <button
    className="ui green button"
    id="submit-destination"
    type="submit"
  >
    Submit
  </button>
</form>
`;
